Assets/GothicVR/Scripts/UI/FontManager.cs

```diff
@@ -2,36 +2,114 @@
 using GVR.Util;
 using System.Collections.Generic;
 using TMPro;
+using GVR.Caches;
+using GVR.Phoenix.Util;
+using UnityEngine;
+using System;
+using UnityEngine.TextCore;
+using System.Reflection;
 
 namespace GVR.Manager
 {
     public class FontManager : SingletonBehaviour<FontManager>
     {
-        public Dictionary<string, TMP_FontAsset> fontDictionary = new(); // Dictionary to store font assets for each tag
+
 
         public void Create()
         {
-            fontDictionary[ConstantsManager.I.MenuFontTag] = GameData.I.GothicMenuFont;
-            fontDictionary[ConstantsManager.I.SubtitleFontTag] = GameData.I.GothicSubtitleFont;
+            TMP_Settings.defaultSpriteAsset = LoadFont("font_old_20_white.FNT");
+            TMP_Settings.defaultFontAsset = GameData.I.EmptyFont;
         }
 
         public void ChangeFont()
         {
             // Get all the TextMeshPro components in the scene
             TMP_Text[] textComponents = FindObjectsOfType<TextMeshProUGUI>();
 
-            // Change the font of the TextMeshPro components based on their tags
             foreach (TMP_Text textComponent in textComponents)
             {
-                if (fontDictionary.TryGetValue(textComponent.tag, out var value))
-                    textComponent.font = value;
-                
-                // On Main Menu text is very small if gothic font is used
-                if(textComponent.tag == ConstantsManager.I.MenuFontTag && GameData.I.GothicMenuFont != null)
+                textComponent.font = GameData.I.EmptyFont;
+
+                // texture is rotated 180 degrees
+                textComponent.rectTransform.localRotation = Quaternion.Euler(180, 0, 0);
+            }
+        }
+
+        public TMP_SpriteAsset LoadFont(string fontName)
```

Comment:
Do you have a mechanism in place, that a font isn't generated twice? I.e. how do you handle caching of this function?